libs/components/src/lib/toggletip/toggletip.spec.ts

```diff
@@ -0,0 +1,162 @@
+			await elementUpdated(element);
+
+			element.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should close when clicked outside', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			document.body.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+
+		it('should close when Escape is pressed', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			document.body.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape', bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+	
+	});
+
+	describe('anchor', () => {
+		it('should accept an HTMLElement as anchor', async () => {
+			element.anchor = anchor;
+			await elementUpdated(element);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should open correctly when its anchor is changed', async () => {
+			const anchor2 = fixture(
+				'<vwc-button id="anchorButton2"></vwc-button>', ADD_TEMPLATE_TO_FIXTURE
+			) as Button;
+	
+			element.anchor = 'anchorButton';
+			await elementUpdated(element);
+
+			element.anchor = 'anchorButton2';
+			await elementUpdated(element);
+			expect(element.open).toEqual(false);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+			expect(element.open).toEqual(false);
+
+			anchor2.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+			expect(element.open).toEqual(true);
+		});
+
+		it('should update the anchor aria-label when none is set', async () => {
+			element.anchor = anchor;
+			await elementUpdated(element);
+
+			expect(anchor.ariaLabel).toEqual(' ; Show more information');
+
+			element.anchor = '';
+			await elementUpdated(element);
+
+			expect(anchor.ariaLabel).toEqual('');
+		});
+
+		it('should update the anchor aria-label when it already has one', async () => {
+			const initialLabel = 'some existing label';
+			
+			anchor.ariaLabel = initialLabel;
+			element.anchor = anchor;
+			await elementUpdated(element);
+
+			expect(anchor.ariaLabel).toEqual(initialLabel + ' ; Show more information');
+
+			element.anchor = '';
+			await elementUpdated(element);
+
+			expect(anchor.ariaLabel).toEqual(initialLabel);
+		});
+	});
+
+	describe('headline', () => {
+		it('should have an headline element only when set', async () => {
+			expect(element.shadowRoot?.querySelectorAll('header.headline')).toHaveLength(0);
+
+			element.headline = 'A title!';
+			await elementUpdated(element);
+
+			expect(element.shadowRoot?.querySelectorAll('header.headline')).toHaveLength(1);
+		});
+	});
```

Comment:
Defer expectations to the end and make it clear without understanding the code what is actually being tested.
 
```suggestion
		it('should have an headline element only when set', async () => {
		  const headlineExistsWithoutHeadlineSet = element.shadowRoot?.querySelectorAll('header.headline').length > 0;

			element.headline = 'A title!';
			await elementUpdated(element);
			const headlineExistsWithHeadlineSet = element.shadowRoot?.querySelectorAll('header.headline').length === 1;

      expect(headlineExistsWithoutHeadlineSet).toEqual(false);
			expect(headlineExistsWithHeadlineSet).toEqual(true);
		});
	});
```

Reply:
this is not an improvement.

Reply:
To me it is :)